### PR TITLE
91706 subscriber name length validation

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.Application/Validators/UpsertSubscriberRequestValidator.cs [hunk 1]
```diff
@@ -9,7 +9,9 @@ public class UpsertSubscriberRequestValidator : AbstractValidator<UpsertSubscrib
         public UpsertSubscriberRequestValidator()
         {
             RuleFor(x => x.EventName).NotEmpty();
-            RuleFor(x => x.SubscriberName).NotEmpty();
+            RuleFor(x => x.SubscriberName).Cascade(CascadeMode.Stop)
+               .NotEmpty()
+               .MaximumLength(50);
```
Comment (chomick) [inline src/CaptainHook.Application/Validators/UpsertSubscriberRequestValidator.cs:14]: It is not the subscriber name that is limited, it is a subscription name in service bus, which is built of event name, subscriber name and few other things

Comment (chomick) [reply src/CaptainHook.Application/Validators/UpsertSubscriberRequestValidator.cs:14]: That's why we need to extract the name building to a separate class so we can reuse it here

Comment (dawidkeshop) [reply src/CaptainHook.Application/Validators/UpsertSubscriberRequestValidator.cs:14]: I've added comment to the story, we use directly subscriber name, we don't append event name or any other things there.

Comment (mszymeshop) [reply src/CaptainHook.Application/Validators/UpsertSubscriberRequestValidator.cs:14]: I guess this checks out @dawidkeshop. The concatenation of event and subscriber is used internally for CH Readers, SB subscribers are just SubscriberNames.

**Review by mszymeshop — APPROVED** [review]

File: src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertSubscriberRequestValidatorTests.cs [hunk 2]
```diff
@@ -94,6 +94,16 @@ public void When_SubscriberIsEmpty_Then_ValidationFails(string invalidString)
             result.ShouldHaveValidationErrorFor(x => x.SubscriberName);
         }
 
+        [Fact, IsUnit]
+        public void When_SubscriberIsTooLong_Then_ValidationFails()
```
Comment (tabman83) [inline src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertSubscriberRequestValidatorTests.cs:98]: When_Subscriber_Name_IsTooLong_Then_ValidationFails

**Review by tabman83 — APPROVED** [review]

File: src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertSubscriberRequestValidatorTests.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertSubscriberRequestValidatorTests.cs]: sure

File: src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertWebhookRequestValidatorTests.cs [hunk 3]
```diff
@@ -63,6 +63,16 @@ public void When_SubscriberIsEmpty_Then_ValidationFails(string invalidString)
             result.ShouldHaveValidationErrorFor(x => x.SubscriberName);
         }
 
+        [Fact, IsUnit]
+        public void When_SubscriberIsTooLong_Then_ValidationFails()
```
[the remark this answers is not in the document]
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Application.Tests/RequestValidators/UpsertWebhookRequestValidatorTests.cs]: sure

**Pull request merged**